packages/cli-kit/src/private/node/components/ConcurrentOutput.tsx

```diff
@@ -0,0 +1,114 @@
+import {OutputProcess} from '../../../output.js'
+import React, {FunctionComponent, useEffect, useState} from 'react'
+import {Static, Text, useApp} from 'ink'
+import stripAnsi from 'strip-ansi'
+import {Writable} from 'node:stream'
+
+export type WritableStream = (process: OutputProcess, index: number) => Writable
+export type RunProcesses = (writableStream: WritableStream, unmountInk: (error?: Error | undefined) => void) => Promise<void>
+
+interface Props {
+  processes: OutputProcess[]
+  runProcesses: RunProcesses
+}
+
+interface Line {
+  color: string
+  value: string
+  prefix: string
+}
+
+/**
+ * Renders output from concurrent processes to the terminal.
+ * Output will be divided in a two column layout, with the left column
+ * containing the process prefix and the right column containing the output.
+ * Every process will be rendered with a different color, up to 4 colors.
+ *
+ * For example running `shopify app dev`:
+ *
+ * ```shell
+ * 2022-10-10 13:11:03 | backend    | npm
+ * 2022-10-10 13:11:03 | backend    |  WARN ignoring workspace config at ...
+ * 2022-10-10 13:11:03 | backend    |
+ * 2022-10-10 13:11:03 | backend    |
+ * 2022-10-10 13:11:03 | backend    | > shopify-app-template-node@0.1.0 dev
+ * 2022-10-10 13:11:03 | backend    | > cross-env NODE_ENV=development nodemon backend/index.js --watch ./backend
+ * 2022-10-10 13:11:03 | backend    |
+ * 2022-10-10 13:11:03 | backend    |
+ *
+ * 2022-10-10 13:11:03 | frontend   |
+ * 2022-10-10 13:11:03 | frontend   | > starter-react-frontend-app@0.1.0 dev
+ * 2022-10-10 13:11:03 | frontend   | > cross-env NODE_ENV=development node vite-server.js
+ * 2022-10-10 13:11:03 | frontend   |
+ * 2022-10-10 13:11:03 | frontend   |
+
+ * 2022-10-10 13:11:03 | backend    | [nodemon] 2.0.19
+ * 2022-10-10 13:11:03 | backend    |
+ * 2022-10-10 13:11:03 | backend    | [nodemon] to restart at any time, enter `rs`
+ * 2022-10-10 13:11:03 | backend    | [nodemon] watching path(s): backend/
+ * 2022-10-10 13:11:03 | backend    | [nodemon] watching extensions: js,mjs,json
+ * 2022-10-10 13:11:03 | backend    | [nodemon] starting `node backend/index.js`
+ * 2022-10-10 13:11:03 | backend    |
+ *
+ * ```
+ */
+const ConcurrentOutput: FunctionComponent<Props> = ({processes, runProcesses}) => {
```

Comment:
I wanted to let the parent decide how to run the processes so that the `ConcurrentOutput` wouldn't have to do so in the `useEffect` hook, but I agree that it's awkward to have both passed in. I'll think about alternatives, but I'd rather preserve the `runProcesses` argument than `processes`.